contracts/token/ERC20/extensions/ERC20Votes.sol

```diff
@@ -38,7 +43,12 @@ abstract contract ERC20Votes is ERC20, Votes {
     function _update(address from, address to, uint256 amount) internal virtual override {
         super._update(from, to, amount);
         if (from == address(0)) {
-            require(totalSupply() <= _maxSupply(), "ERC20Votes: total supply risks overflowing votes");
+            uint256 supply = totalSupply();
+            uint256 cap = _maxSupply();
+            if (supply > cap) {
+                // Exceeding supply introduces a risk of votes overflowing
+                revert ERC20ExceededCap(supply, cap);
+            }
```

Comment:
Looking at that, I think we could simplify the code by doing

```solidity
abstract contract ERC20Votes is ERC20, ERC20Capped(type(uint224).max), Votes {
```

(for another PR though)

Reply:
Interesting finding, I'd also leave a TODO here for searching if there's historical information of why we didn't do this initially.

Maybe @frangio has an input

Reply:
I would not reuse ERC20Capped, my reason is that it's such little code that it's better to make it straightforward.

Reply:
Also, if we use ERC20Capped and some user wants to use ERC20Capped with a custom cap on top of ERC20Votes, that would be a mess.

I'm starting to think we should remove ERC20Capped.

Reply:
I think we agreed with @frangio of not reusing it, and based on the conversation we had in today's sync, we're considering removing `ERC20Capped`, and we're keeping track of that aside, so I'm solving this thread since I think the discussion was moved to the removal tracking.

Please reopen if there's something else to add in this thread.